packages/aws-cdk-lib/aws-ecr/lib/auto-delete-images-handler/index.ts

```diff
@@ -40,14 +40,23 @@ async function emptyRepository(params: ECR.ListImagesRequest) {
   const listedImages = await ecr.listImages(params).promise();
 
   const imageIds = listedImages?.imageIds ?? [];
+  const imageIdsTagged = imageIds.filter((imageId) => 'imageTag' in imageId) ?? [];
```

Comment:
Looks like you would be processing the tagged images twice. Can we instead create two lists that don't have overlap?

Maybe something like

```ts
const imageIds = [];
const imageIdsTagged = [];
(listedImages.imageIds ?? []).forEach(imageId => {
  if ('imageTag' in imageId) {
    imageIdsTagged.push(imageId);
  } else {
    imageIds.push(imageId);
  }
});
```